### PR TITLE
wlprop: init at 2022-08-18

### PR DESCRIPTION
**Pull request by SebTM**
###### Description of changes

init wlprop at 2022-08-18

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/wayland/wlprop/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{ fetchgit, gawk, jq, lib, makeWrapper, slurp, stdenv, sway }:
+
+stdenv.mkDerivation rec {
+  pname = "wlprop";
+  version = "2022-08-18";
```
Comment (SuperSandro2000) [inline pkgs/tools/wayland/wlprop/default.nix:5]:
```suggestion
  version = "unstable-2022-08-18";
```

File: pkgs/tools/wayland/wlprop/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  nativeBuildInputs = [ makeWrapper ];
+  buildInputs = [ gawk jq slurp sway ];
+
+  dontBuild = true;
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    install -Dm755 wlprop.sh $out/bin/wlprop
```
Comment (SuperSandro2000) [inline pkgs/tools/wayland/wlprop/default.nix:21]:
```suggestion
    install -dDm755 wlprop.sh $out/bin/wlprop
```

File: pkgs/tools/wayland/wlprop/default.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    wrapProgram "$out/bin/wlprop" \
+      --prefix PATH : "$out/bin:${lib.makeBinPath [ gawk jq slurp sway ]}"
+
+    runHook postInstall
+  '';
+  passthru.scriptName = " wlprop.sh";
```
Comment (SuperSandro2000) [inline pkgs/tools/wayland/wlprop/default.nix:27]:
```suggestion
  passthru.scriptName = "wlprop.sh";
```

File: pkgs/tools/wayland/wlprop/default.nix [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  version = "2022-08-18";
+
+  src = fetchgit {
+    url = "https://gist.github.com/f313386043395ff06570e02af2d9a8e0";
+    rev = "758c548bfb4be5b437c428c8062b3987f126f002";
+    sha256 = "sha256-ZJ9LYYrU2cNYikiVNTlEcI4QXcoqfl7iwk3Be+NhGG8=";
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+  buildInputs = [ gawk jq slurp sway ];
```
[the remark this answers is not in the document]
Comment (SebTM) [reply pkgs/tools/wayland/wlprop/default.nix]: Is this only relevant for buildInputs or also for native?

**Comment by SebTM** [comment]
Thanks for your review, adjusted the PR accordingly.

File: pkgs/tools/wayland/wlprop/default.nix (hunk 2 above)
Comment (SebTM) [reply pkgs/tools/wayland/wlprop/default.nix]: Does not build for me, but was able to drop mkdir without adding "d" without any issues:
```
install: cannot create directory 'wlprop.sh': File exists
```

**Pull request merged**